packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue

```diff
@@ -8,36 +8,41 @@
             has-outline>
             <form @submit.prevent="onFormSubmit">
                 <div
-                    v-for="{ email, key, sms } in preferences"
+                    v-for="{ key, emailEnabled, emailValue, smsEnabled, smsValue } in preferences"
                     :key="key">
-                    <h2 :class="$style['c-contactPreferences-subtitle']">
+                    <h2
+                        :class="$style['c-contactPreferences-subtitle']">
                         {{ $t(`${key}.subtitle`) }}
                     </h2>
-                    <fieldset :class="$style['c-contactPreferences-fieldset']">
+                    <fieldset
+                        :class="$style['c-contactPreferences-fieldset']">
                         <label>
                             <input
                                 type="checkbox"
-                                :disabled="!email.enabled"
-                                :checked="email.value">
+                                :data-test-id="`contactPreferences-${key}-checkbox`"
+                                :disabled="!emailEnabled"
+                                :checked="emailValue"
+                                @change="editPreferenceValue(key, Object.keys({ emailValue })[0], $event.target.checked)">
                             <span
                                 :class="{
-                                    [$style['c-contactPreferences-labelText--disabled']]: !email.enabled
+                                    [$style['c-contactPreferences-labelText--disabled']]: !emailEnabled
                                 }">
                                 {{ $t(`${key}.email`) }}
                                 <template v-if="$te(`${key}.emailDescription`)">
                                     <br>({{ $t(`${key}.emailDescription`) }})
                                 </template>
                             </span>
                         </label>
-
                         <label>
                             <input
                                 type="checkbox"
-                                :disabled="!sms.enabled"
-                                :checked="sms.value">
+                                :data-test-id="`contactPreferences-${key}-checkbox`"
+                                :disabled="!smsEnabled"
+                                :checked="smsValue"
+                                @change="editPreferenceValue(key, Object.keys({ smsValue })[0], $event.target.checked)">
                             <span
                                 :class="{
-                                    [$style['c-contactPreferences-labelText--disabled']]: !sms.enabled
+                                    [$style['c-contactPreferences-labelText--disabled']]: !smsEnabled
```

Comment:
Same here as above `isSmsEnabled`

Reply:
Cool but can I do this on a follow up ticket only because it means changing the mapper + tests (already in upstream)?

Reply:
Yep – happy for this to happen in a separate PR.